python/pyarrow/hdfs.py

```diff
@@ -128,11 +128,18 @@ def _maybe_set_hadoop_classpath():
         return
 
     if 'HADOOP_HOME' in os.environ:
-        hadoop_bin = '{0}/bin/hadoop'.format(os.environ['HADOOP_HOME'])
+        find_args = ('find', os.environ['HADOOP_HOME'], '-name', '*.jar')
+        find = subprocess.Popen(find_args, stdout=subprocess.PIPE)
+        xargs_echo = subprocess.Popen(('xargs', 'echo'),
+                                      stdin=find.stdout,
+                                      stdout=subprocess.PIPE)
+        classpath = subprocess.check_output(('tr', "' '", "':'"),
+                                            stdin=xargs_echo.stdout)
```

Comment:
This will only work on Unixy systems, though. Can you factor this into a function that can be called here only when `sys.platform != 'win32'`?

Reply:
Sure thing, who runs hadoop on windows? ;)

Reply:
You'd be surprised :) The Hadoop client is supported on Windows; that being said some users have reported having issues getting this library to work there, but I would like to see it supported